### DISCUSSION TITLE
Unit Testing Doc: explain evaluation_interval

### DISCUSSION THREAD

**Discussion by michal-sankot** [prometheus/prometheus#12060] [Q&A] [answered]
### Proposal

Describe meaning of `evaluation_interval` in https://github.com/prometheus/prometheus/blob/main/docs/configuration/unit_testing_rules.md.
Currently it has no comment at all and it's unclear what it refers to.

**Comment by roidelapluie** [comment]
In the unit test file, the `evaluation_interval` field is used to specify the evaluation interval for the rules being tested. It determines how often the rules will be evaluated by Promtool. Any tests executed within the test file will be executed against the rules that are evaluated at the interval specified by `evaluation_interval`.

cc @dgl do you see clarification needed for the docs?

**Comment by michal-sankot** [comment]
alright, it's clearer now, thanks.

there are actually 3 attributes that relate to time or time intervals in some way:
- root `evaluation_interval`
- test `interval`
- alert_test_case `eval_time`

so do i understand it right, that
- root `evaluation_interval` - refers to how often promtool checks what alerts are fired
- test `interval` - refers to time unit of input_series
- alert_test_case `eval_time` - refers to point in time where we check what alerts are fired
?

and so if i define `evaluation_interval` to 1h and `interval` is the default 1 minute
- i can set input series values to whatever but what would really count would be their values at 60 minutes multiples
- it would make no sense to set `eval_time`s to anything else then multiples of whole hours 
do i get it right?

**Comment by roidelapluie** [comment, marked as answer]
You are right, however:

> i can set input series values to whatever but what would really count would be their values at 60 minutes multiples

That depends on the PromQL query (_over _time, offset) can change that.

In general you want evaluation interval to e.g. 1 minute, to match Prometheus default behavior.
